scapy/contrib/rpl.py

```diff
@@ -0,0 +1,312 @@
+# This file is part of Scapy.
+# See http://www.secdev.org/projects/scapy for more information.
+#
+# Scapy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Scapy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Scapy.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Copyright (C) 2020 Rahul Jadhav <nyrahul@gmail.com>
+
+# RFC 6550
+# scapy.contrib.description = Routing Protocol for LLNs (RPL)
+# scapy.contrib.status = loads
+
+"""
+RFC 6550 - Routing Protocol for Low-Power and Lossy Networks (RPL)
+draft-ietf-roll-efficient-npdao-17 - Efficient Route Invalidation
+"""
+
+from scapy.packet import Packet, bind_layers
+from scapy.fields import ByteEnumField, ByteField, IP6Field, ShortField, \
+    BitField, BitEnumField, FieldLenField, StrLenField, IntField
+from scapy.layers.inet6 import RPL, icmp6ndraprefs, _IP6PrefixField
+
+
+# https://www.iana.org/assignments/rpl/rpl.xhtml#mop
+RPLMOP = {0: "No Downward routes",
+          1: "Non-Storing",
+          2: "Storing with no multicast support",
+          3: "Storing with multicast support",
+          4: "P2P Route Discovery"}
+
+
+# https://www.iana.org/assignments/rpl/rpl.xhtml#control-message-options
+RPLOPTSSTR = {0: "Pad1",
+              1: "PadN",
+              2: "DAG Metric Container",
+              3: "Routing Information",
+              4: "DODAG Configuration",
+              5: "RPL Target",
+              6: "Transit Information",
+              7: "Solicited Information",
+              8: "Prefix Information Option",
+              9: "Target Descriptor",
+              10: "P2P Route Discovery"}
+
+
+class _RPLGuessOption(Packet):
+    name = "Dummy RPL Option class that implements guess_payload_class()"
```

Comment:
Minor detail but this docstrings is now wrong

Reply:
Fixed the docstring. Thanks